crates/ironrdp-svc/src/lib.rs

```diff
@@ -85,22 +85,77 @@ pub enum CompressionCondition {
     Always,
 }
 
+/// A static virtual channel.
+#[derive(Debug)]
+pub struct StaticVirtualChannel {
+    channel_processor: Box<dyn StaticVirtualChannelProcessor>,
+    chunk_processor: ChunkProcessor,
+}
+
+impl StaticVirtualChannel {
+    pub fn new<T: StaticVirtualChannelProcessor + 'static>(channel_processor: T) -> Self {
+        Self {
+            channel_processor: Box::new(channel_processor),
+            chunk_processor: ChunkProcessor::new(),
+        }
+    }
+
+    pub fn channel_name(&self) -> ChannelName {
+        self.channel_processor.channel_name()
+    }
+
+    pub fn compression_condition(&self) -> CompressionCondition {
+        self.channel_processor.compression_condition()
+    }
+
+    /// Processes a payload received on the virtual channel. Returns a vector of PDUs to be sent back
+    /// to the server. If no PDUs are to be sent, an empty vector is returned.
+    pub fn process(&mut self, payload: &[u8]) -> PduResult<Vec<SvcMessage>> {
+        if let Some(payload) = self.dechunkify(payload)? {
+            return self.channel_processor.process(&payload);
+        }
+
+        Ok(Vec::new())
+    }
+
+    pub fn chunkify(messages: Vec<SvcMessage>) -> PduResult<Vec<WriteBuf>> {
+        ChunkProcessor::chunkify(messages, CHANNEL_CHUNK_LENGTH)
+    }
+
+    pub fn is_drdynvc(&self) -> bool {
+        self.channel_processor.is_drdynvc()
+    }
+
+    pub fn channel_processor_downcast_ref<T: StaticVirtualChannelProcessor + 'static>(&self) -> Option<&T> {
+        self.channel_processor.as_any().downcast_ref()
+    }
+
+    pub fn channel_processor_downcast_mut<T: StaticVirtualChannelProcessor + 'static>(&mut self) -> Option<&mut T> {
+        self.channel_processor.as_any_mut().downcast_mut()
+    }
+
+    fn dechunkify(&mut self, payload: &[u8]) -> PduResult<Option<Vec<u8>>> {
+        self.chunk_processor.dechunkify(payload)
+    }
+}
+
 /// A type that is a Static Virtual Channel
 ///
 /// Static virtual channels are created once at the beginning of the RDP session and allow lossless
 /// communication between client and server components over the main data connection.
 /// There are at most 31 (optional) static virtual channels that can be created for a single connection, for a
 /// total of 32 static channels when accounting for the non-optional I/O channel.
-pub trait StaticVirtualChannel: AsAny + fmt::Debug + Send + Sync {
-    /// Returns the name of the `StaticVirtualChannel`
+pub trait StaticVirtualChannelProcessor: AsAny + fmt::Debug + Send + Sync {
+    /// Returns the name of the `StaticVirtualChannelProcessor`
```

Comment:
Maybe this should refer to the broader concept of static virtual channel here
```suggestion
    /// Returns the name of the static virtual channel for this processor
```